### PR TITLE
Java doc

### PR DESCRIPTION
**Pull request by augnor**
Added lots of java documentation and a few changes like moving model.update in front of  view.update in game

### CODE REVIEW

File: core/src/model/IPlayerCharacter.java [hunk 1]
```diff
@@ -10,16 +10,38 @@
 import model.rewards.Reward;
 import model.rewards.Tweak;
 
+/**
+ * Represents a playerCharacter
+ */
 public interface IPlayerCharacter extends ILivingEntity {
 
+    Direction getDirection();
```
Comment (pkasinski) [inline core/src/model/IPlayerCharacter.java]: This method should be removed from IPlayerCharacter.

File: core/src/model/IPlayerCharacter.java [hunk 2]
```diff
@@ -10,16 +10,38 @@
...
+
     int getExperience();
 
     int getLevel();
 
+    float getCurrentHealth();
+
+    float getSpeed();
+
+    float getDamage();
```
Comment (pkasinski) [inline core/src/model/IPlayerCharacter.java]: These methods should also be removed.

File: core/src/model/IEntity.java [hunk 3]
```diff
@@ -16,5 +19,10 @@ public interface IEntity {
 
     World<IEntity> getWorld();
 
+    /**
+     * method
```
Comment (pkasinski) [inline core/src/model/IEntity.java]: A description is missing.

File: core/src/model/Entity.java [hunk 4]
```diff
@@ -23,6 +23,15 @@ public void setCollisionResponse(CollisionFilter collisionResponse) {
         this.collisionResponse = collisionResponse;
     }
 
+    /**
+     * Abstract constructor
```
Comment (pkasinski) [inline core/src/model/Entity.java]: Abstract constructor? I think it should be written which object is created instead.

File: core/src/application/Game.java [hunk 5]
```diff
@@ -5,6 +5,9 @@
 import view.View;
 import com.badlogic.gdx.ApplicationAdapter;
 
+/**
+ * Runs the main loop of the playable game, in other words the ingame loop
```
Comment (pkasinski) [inline core/src/application/Game.java]: Looks like a description of a method.

File: core/src/model/Entity.java [hunk 6]
```diff
@@ -23,6 +23,15 @@ public void setCollisionResponse(CollisionFilter collisionResponse) {
...
+     * @param x entity x coordinate position
+     * @param y entity y coordinate position
+     * @param height entity height
+     * @param width entity width
+     * @param damage entity damage
+     * @param world the world entity resides in
```
Comment (pkasinski) [inline core/src/model/Entity.java]: It should probably be mentioned that world contains information about collision boxes.

File: core/src/model/TiledMapCache.java [hunk 7]
```diff
@@ -142,6 +142,9 @@ private void importCollisionLayer() {
         }
     }
 
+    /**
+     * Fetches a list of spawnpoints from a map if it has a baked in spawnlayer, otherwise uses default spawnpoints
```
Comment (pkasinski) [inline core/src/model/TiledMapCache.java:146]: If it has a baked?

File: core/src/view/HUD.java [hunk 8]
```diff
@@ -76,10 +75,13 @@ public HUD(SpriteBatch spriteBatch, Model model) {
         this.model = model;
     }
 
+    /**
+     * update updates everything shown invthe HUD.
```
Comment (pkasinski) [inline core/src/view/HUD.java]: Typo.

File: core/src/model/Model.java [hunk 9]
```diff
@@ -44,6 +50,10 @@ public IPlayerCharacter getPlayer(){
         return player;
     }
 
+    /**
+     * Spawns monsters if there are too few, moves monsters toward the player, checks for player levelup
```
Comment (pkasinski) [inline core/src/model/Model.java]: This description could possibly be simplified.

**Review by pkasinski — CHANGES_REQUESTED** [review]

File: core/src/model/Entity.java (hunk 4 above)
Comment (rojtekin) [reply core/src/model/Entity.java]: should be "Abstract constructor for Entity" atleast

File: core/src/model/Model.java (hunk 9 above)
Comment (rojtekin) [reply core/src/model/Model.java]: description too detailed, should be along the lines of "updates the state of the model"

File: core/src/model/Sword.java [hunk 10]
```diff
@@ -4,6 +4,9 @@
 
 public class Sword extends PlayerWeapon{
 
+    /**
+     * Creates an instance of Sword
```
Comment (rojtekin) [inline core/src/model/Sword.java]: Constructs*

File: core/src/view/HUD.java [hunk 11]
```diff
@@ -30,8 +30,8 @@ public class HUD {
 
     private Image healthColor;
 
-    //The code under is a way to make a certain color since libgdx somehow doesnt know rgb values needed
-    //for a square (and also make it an actor so I can send it back and not draw directly).
+    //The code under is a way to make a certain color since libgdx somehow doesnt know rgb values (or I missed it)
+    //for a square (and also make it an actor so I can add it to stage).
```
Comment (rojtekin) [inline core/src/view/HUD.java]: in line comments should be removed

**Review by rojtekin — APPROVED** [review]
LGTM for the most part, some places needs fixing check my comments

File: core/src/application/Game.java (hunk 5 above)
Comment (augnor) [reply core/src/application/Game.java]: Application the create and connects medel, view, controller

**Review by pkasinski — APPROVED** [review]

**Pull request merged**